com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/PollStageAppStatusExecution.java

```diff
@@ -41,67 +41,70 @@ public AsyncExecutionState execute(ExecutionWrapper execution) {
 
-            if (!state.getState().equals(PackageState.STAGED)) {
-                setStagingLogs(state, execution);
+            if (!state.getState()
+                      .equals(PackageState.STAGED)) {
+                setStagingLogs(state, execution, app.getName(), processLoggerProvider);
                 
                 return checkStagingState(execution, app, state);
             }
 
             execution.getStepLogger()
-                .info(Messages.APP_STAGED, app.getName());
+                     .info(Messages.APP_STAGED, app.getName());
 
             UUID appId = client.getApplication(app.getName())
-                .getMeta()
-                .getGuid();
+                               .getMeta()
+                               .getGuid();
             
             applicationStager.bindDropletToApp(execution, appId, client);
 
             return AsyncExecutionState.FINISHED;
         } catch (CloudOperationException coe) {
             CloudControllerException e = new CloudControllerException(coe);
             execution.getStepLogger()
-                .error(e, Messages.ERROR_STAGING_APP_1, app.getName());
+                     .error(e, Messages.ERROR_STAGING_APP_1, app.getName());
             throw e;
         } catch (SLException e) {
             execution.getStepLogger()
-                .error(e, Messages.ERROR_STAGING_APP_1, app.getName());
+                     .error(e, Messages.ERROR_STAGING_APP_1, app.getName());
             throw e;
         }
     }
 
     private AsyncExecutionState checkStagingState(ExecutionWrapper execution, CloudApplication app, StagingState state) {
-        if (state.getState().equals(PackageState.FAILED)) {
+        if (state.getState()
+                 .equals(PackageState.FAILED)) {
             // Application staging failed
             String message = format(Messages.ERROR_STAGING_APP_2, app.getName(), state.getError());
             execution.getStepLogger()
-                .error(message);
+                     .error(message);
             return AsyncExecutionState.ERROR;
         }
         // Application not staged yet, wait and try again unless it's a timeout.
         return AsyncExecutionState.RUNNING;
     }
     
-    private void setStagingLogs(StagingState state, ExecutionWrapper execution) {
+    private void setStagingLogs(StagingState state, ExecutionWrapper execution, String appName, ProcessLoggerProvider processLoggerProvider) {
         if (state.getStagingLogs() != null) {
             StagingLogs logs = state.getStagingLogs();
             String stagingLogs = logs.getLogs();
             int offset = logs.getOffset();
             
             stagingLogs = new XMLValueFilter(stagingLogs).getFiltered();
             execution.getStepLogger()
-                .debug(stagingLogs);
+                     .debug(stagingLogs);
+            StepsUtil.appLog(execution.getContext(), appName, stagingLogs, LOGGER, processLoggerProvider);
```

Comment:
Is this about add staging logs into application log file?

Reply:
yes, and actually this whole if is applicable only to xs because there is no staging logs in the state object for cf